Examples/Sources/MacroExamples/Implementation/Expression/FontLiteralMacro.swift

```diff
@@ -16,13 +16,13 @@ import SwiftSyntaxMacros
 /// Implementation of the `#fontLiteral` macro, which is similar in spirit
 /// to the built-in expressions `#colorLiteral`, `#imageLiteral`, etc., but in
 /// a small macro.
-public struct FontLiteralMacro: ExpressionMacro {
+public enum FontLiteralMacro: ExpressionMacro {
```

Comment:
Not dived that much into macros. 
But is `enum` the way to use macros ?

Reply:
It can be either, really. The enum type just needs to provide static methods.

Reply:
Okay, so it's same when developers use an `enum` for name spaces?

What is then the "preferred" way to do it? 
What does documentation and WWDC videos do?

It's more to do what they do, so people reading macro for the first time and comparing to the videos, and get a bit confused.

Reply:
Since the declaration needs to provide static methods, an `enum` appears to be the best choice. I assume there won't be a scenario where a user would want to create an instance of a declaration implementing macro protocols.

I haven't yet caught up on the WWDC videos (they're still on my to-watch list 🙈), but from what I've seen in the documentation, there's only one instance where a `struct` is used. Moreover, the available documentation on macros seems somewhat sparse; it doesn't appear to be very thorough.

By the way, if I find some time, I'm gonna to write DocC tutorials about macros. That way, we could make it the primary entry point for those interested in learning about macros.

Reply:
@DougGregor Do you have any opinions here?

Reply:
Only static methods are used to implement macros, so in essence it doesn't matter what kind of type you use. I'm fine with `enum`, as a caseless enum makes it clear that there will never be an instance.